schema/graph.json

```diff
@@ -0,0 +1,236 @@
+                    "$ref": "#/definitions/start_time"
+                },
+                "end_time": {
+                    "$ref": "#/definitions/end_time"
+                },
+                "cloning_rate": {
+                    "$ref": "#/definitions/rate"
+                },
+                "selfing_rate": {
+                    "$ref": "#/definitions/rate"
+                },
+                "size_function": {
+                    "type": "string"
+                }
+            },
+            "additionalProperties": false
+        },
+        "deme": {
+            "type": "object",
+            "properties": {
+                "id": {
+                    "$ref": "#/definitions/id"
+                },
+                "description": {
+                    "type": "string"
+                },
+                "ancestors": {
+                    "type": "array",
+                    "minItems": 1,
+                    "items": {
+                        "$ref": "#/definitions/id"
+                    }
+                },
+                "proportions": {
+                    "type": "array",
+                    "minItems": 1,
+                    "items": {
+                        "$ref": "#/definitions/proportion"
+                    }
+                },
+                "initial_size": {
+                    "$ref": "#/definitions/size"
+                },
+                "final_size": {
+                    "$ref": "#/definitions/size"
+                },
+                "start_time": {
+                    "$ref": "#/definitions/start_time"
+                },
+                "end_time": {
+                    "$ref": "#/definitions/end_time"
+                },
+                "cloning_rate": {
+                    "$ref": "#/definitions/rate"
+                },
+                "selfing_rate": {
+                    "$ref": "#/definitions/rate"
+                },
+                "epochs": {
+                    "type": "array",
+                    "minItems": 1,
+                    "items": {
+                        "$ref": "#/definitions/epoch"
+                    }
+                }
+            },
+            "additionalProperties": false,
+            "required" : ["id"]
+        },
+        "pulse": {
+            "type": "object",
+            "properties": {
+                "source": {
+                    "$ref": "#/definitions/id"
+                },
+                "dest": {
+                    "$ref": "#/definitions/id"
+                },
+                "time": {
+                    "type": "number",
+                    "exclusiveMinimum": 0,
+                    "exclusiveMaximum": Infinity
+                },
+                "proportion": {
+                    "$ref": "#/definitions/proportion"
+                }
+            },
+            "additionalProperties": false,
+            "required" : ["source", "dest", "time", "proportion"]
+        },
+        "symmetric_migration": {
```

Comment:
I think the asymmetric and symmetric migrations are swapped here - asymmetric takes source and dest, and while symmetric a list of at least two demes